### PR TITLE
Teleop panel

### PR DESCRIPTION
**Pull request by jreyesr**
_(First time contributor, so apologies for any mistakes)_

## Summary

This PR adds a Teleoperation panel which operates in a similar way to the [teleop_twist_keyboard](http://wiki.ros.org/teleop_twist_keyboard) node. It can be controlled by clicking buttons or with the arrow keys.

Teleop panel

The main motivation of this change would be its use in remotely controlled (i.e., non-autonomous) robots. Such robots can't currently be easily controlled via Webviz (at least, I don't know of a simple way), and it would be nice to have the teleoperation interface next to, for example, an Image panel showing the robot cameras, or a Map panel showing the robot location (or both!).

This PR started by copying the Publish panel, since its functionality is similar. The major difference is that the Teleop panel sends specific messages to a fixed (configurable) topic with a fixed (non-configurable) datatype, instead of arbitrary mesages to any topic. Unnecessary settings were deleted, and the UI was edited to only contain a few buttons which trigger a message publication via a Publisher component. A global KeyListener element was also added, to allow keypresses to also trigger the messages.

## Test plan

The new panel was manually tested on a local installation, with ROS running on a Docker container and the Websockets server exposed via port 9090. This change makes no sense when using local or remote bags.
The test files from the Publish panel were also copied, adapted and run. All tests pass successfully.

## Versioning impact

This PR should only require a minor semver update. 

## Addendum: Known limitations & questions
* At present, the panel suffers from issue https://github.com/cruise-automation/webviz/issues/534 which is also present in the Publish panel, since they both use the same Publisher component. However, unlike the Publish panel, this one displays an error message over the panel, with a "Reload Panel" button. Clicking on the button (seems to) fix the error. A fix to the linked issue should (?) remove this bug.
* The panel only allows four directions (up, down, left, right) plus a Stop command. More complex robots can use up to nine keys  for direction, plus a few extra to change rotation and translation speed. Should those additional keys be implemented?
* Currently, the four directions are bound to the arrow keys, and Stop is bound to the Space key. Should the keybindings be configurable? If so, how? Maybe a dropdown for every command, with a list of valid keys?

### CODE REVIEW

File: packages/webviz-core/src/panels/Teleop/index.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+    const { datatypeNames, parsedObject, error, pressing } = this.state;
+    const canPublish = capabilities.includes(PlayerCapabilities.advertise);
+
+    return (
+      <Flex col style={{ height: "100%", padding: "12px" }} onKeyPress={this.handleKey}>
+        {topicName && (
+          <Publisher ref={this._publisher} name="Publish" topic={topicName} datatype="geometry_msgs/Twist" />
+        )}
+        
+        <KeyListener global keyDownHandlers={this._keyHandlers} keyUpHandlers={this._keyHandlers} />
```
Comment (vidaaudrey) [inline packages/webviz-core/src/panels/Teleop/index.js:248]: Let's keep the key listerner local? We'd want to preserve the arrow keys for global usage, e.g. currently left/right arrow will seek back/forward 100ms.

File: packages/webviz-core/src/panels/Teleop/index.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    } else {
+      parsedObject = parsedAny;
+    }
+  } catch (e) {
+    error = value ? e.message : "";
+  }
+  return { error, parsedObject };
+}
+
+class Teleop extends React.PureComponent<Props, PanelState> {
```
Comment (vidaaudrey) [inline packages/webviz-core/src/panels/Teleop/index.js:105]: Could you convert this component to functional component. We've converted the majority of components but still a few left. Would be nice to use the new syntax.

File: packages/webviz-core/src/panels/Teleop/index.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+    return (
+      <>
+        <Item>
+          <PanelToolbarLabel>Button color (rgba or hex)</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.buttonColor}
+            onChange={(event) => {
+              saveConfig({ buttonColor: event.target.value });
```
Comment (vidaaudrey) [inline packages/webviz-core/src/panels/Teleop/index.js:212]: Curious what's your use case for customizing button color?

File: packages/webviz-core/src/panels/Teleop/index.js [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+            placeholder="rgba(1,1,1,1) or #FFFFFF"
+          />
+        </Item>
+        <Item>
+          <PanelToolbarLabel>Topic name</PanelToolbarLabel>
+          <PanelToolbarInput
+            type="text"
+            value={config.topicName}
+            onChange={(event) => {
+              saveConfig({ topicName: event.target.value });
```
Comment (vidaaudrey) [inline packages/webviz-core/src/panels/Teleop/index.js:223]: Would need some data validation to ensure the saved topic name is valid

**Review by vidaaudrey — COMMENTED** [review]
Thanks for your contribution.  Added a few comments. There are some lint errors in CI. If you are using vscode, you could install `eslint` to auto format the code. Let me know if you need any help.

File: packages/webviz-core/src/panels/Teleop/index.js (hunk 1 above)
Comment (jreyesr) [reply packages/webviz-core/src/panels/Teleop/index.js]: Done. It makes things a bit awkward, though, you now need to click on any button before the key presses are sent to the component. I guess it can't be avoided, you need to give focus to the component if it's not global.

File: packages/webviz-core/src/panels/Teleop/index.js (hunk 2 above)
Comment (jreyesr) [reply packages/webviz-core/src/panels/Teleop/index.js]: Done (I think so, at least, I don't really work on React and so I had to google what a functional component is). Can you confirm that I actually converted the component to functional style, please?

File: packages/webviz-core/src/panels/Teleop/index.js (hunk 3 above)
Comment (jreyesr) [reply packages/webviz-core/src/panels/Teleop/index.js]: Nothing actually, it's a remnant from the Publish panel from which I pulled the original code. I deleted it.

File: packages/webviz-core/src/panels/Teleop/index.js (hunk 4 above)
Comment (jreyesr) [reply packages/webviz-core/src/panels/Teleop/index.js:119]: Not sure about what would be required to validate the topic. Would it require checking that the topic name exists and it's of the correct datatype? Doing that would limit use a lot, I think, since it may very well happen that the topic doesn't "exist" since no other nodes have ever published to it, and the Teleop panel will be the only one publishing. Any pointers? 
Perhaps a) if it doesn't exist, just accept it, and b) if it already exists, ensure that the datatype is actually "geometry_msgs/Twist"?

**Comment by jaguardo** [comment]
@jreyesr I tried the changes (granted, I might be missing something as well!) but I couldn't get the pop up panel for the add panel or change panel to list the teleop as an option... is there something else that has to changes after making those changes? thanks!  From the pictures, I love the functionality, great!

**Comment by jreyesr** [comment]
@jaguardo Weird. I just ran `build-static-webviz` from scratch and it does show me the new panel (on Chrome, Linux). Also tested it on Edge 87.0.664.60, and it works nicely. No idea where that error came from, I do use a KeyListener, but I use it in the same way that other panels (e.g., the Plot panel) use it.

**Comment by jreyesr** [comment]
@vidaaudrey I added a few commits fixing the linter errors and converting to a functional component. CI still fails, but the error now happens when attempting to upload something (screenshots, I think) to S3; the error says `Missing credentials in config`. 
Still unsure about the data validation (see https://github.com/cruise-automation/webviz/pull/561#discussion_r544742818).

**Comment by vidaaudrey** [comment]
@jreyesr we are aware of the CI issue and we have a task to migrate the screenshot testing to GCP soon. At the same time, I just made a copy of your branch at https://github.com/cruise-automation/webviz/pull/567 so you could check the CI output in 1 hour or so.

File: packages/webviz-core/src/panels/Teleop/index.js (hunk 4 above)
Comment (vidaaudrey) [reply packages/webviz-core/src/panels/Teleop/index.js:125]: Sorry about the late reply. A simple validation like this should be ok:
- name is not empty 
- name should start with `/`
- name should be at least 2 characters 

You could refer to some of the validators here: https://github.com/cruise-automation/webviz/blob/master/packages/webviz-core/shared/validators.js#L102-L107

Comment (jreyesr) [reply packages/webviz-core/src/panels/Teleop/index.js:158]: I've added some validation on 671f133. It shows an error message on the panel if those conditions are not met. It also disables all the buttons and keypresses do nothing.

**Comment by amacneil** [comment]
For anyone watching this issue - we recently added a teleop panel to [Foxglove Studio](https://github.com/foxglove/studio) (webviz fork) in [#1429](https://github.com/foxglove/studio/pull/1429).

**Comment by pringithub** [comment]
> @jaguardo Weird. I just ran `build-static-webviz` from scratch and it does show me the new panel (on Chrome, Linux). Also tested it on Edge 87.0.664.60, and it works nicely. No idea where that error came from, I do use a KeyListener, but I use it in the same way that other panels (e.g., the Plot panel) use it.

@jreyesr Doesn't work for me either. I did a clean build and it doesn't show up with `serve-static-webviz`. Something I'm doing wrong? It seems
